### PR TITLE
Add the ability to inject interfaces into Minecraft classes

### LINKED ISSUE #82: Injected interfaces based on Asm
I had an idea for a long time to write a small feature for archloom that would look for all the mixins and analyze them with asm for interface impl, I made such an implementation and it even works, however I am not very familiar with the archloom structure and I don't know how to explain it, you can see the implementation here:
- https://github.com/architectury/architectury-loom/compare/dev/1.6...RedlanceMinecraft:cringe-loom:dev/1.6

Well, I'm sure you guys can come up with something better than my shit code ❤️
I also know you have some thing for analyzing uncompiled java code, so here goes

### PR DESCRIPTION
**Pull request by shartte**
### Interface Injection

Interface injection is an advanced feature allowing mods to add additional interfaces to Minecraft classes and interfaces
at development time. This feature requires that mods use ASM or Mixins to make the same extensions at runtime.

To use this feature, place an [interface injection data-file](https://github.com/neoforged/JavaSourceTransformer?tab=readme-ov-file#interface-injection) in your project and configure the `interfaceInjectionData` property to include it.
Since this feature only applies at development time, you do not need to include this data file in your jar.

> [!IMPORTANT]
> This feature only applies at development time. You need to use Mixins or Coremods to make it work at runtime.

`build.gradle`
```groovy
neoForge {
    interfaceInjectionData = files("interfaces.json")
}
```

`interfaces.json`
```json filename="interfaces.json"
{
  "net/minecraft/world/item/ItemStack": [
    "testproject/FunExtensions"
  ]
}
```

In addition, you can add additional data-files to the `interfaceInjectionData` configuration using normal
Project dependency syntax in your dependencies block.

Fixes #82

### CODE REVIEW

**Review by Matyrobbrt — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -368,6 +368,65 @@ the centrally managed repositories for that project.
...
+
+`build.gradle`
+```groovy
+neoForge {
+    interfaceInjectionData = files("interfaces.json")
+}
+```
+
+`interfaces.json`
+```json filename="interfaces.json"
```
Comment (Technici4n) [inline README.md:419]: Cool filename trick!

File: src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java [hunk 2]
```diff
@@ -187,6 +186,16 @@ public void apply(Project project) {
             });
         });
 
+        // Create a configuration for grabbing interface injection data
+        var interfaceInjectionData = configurations.create("interfaceInjectionData", spec -> {
+            spec.setDescription("Interface injection data add interface extend/implements clauses to Minecraft classes at development time");
```
Comment (Technici4n) [inline src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java:191]: Not quite English I think. :smile:

**Review by Technici4n — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Matyrobbrt) [reply README.md]: It works in the docs, unfortunate that it doesn't on GH.

Comment (Technici4n) [reply README.md]: Lol ok 😅

**Pull request merged**